bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -1259,7 +1261,17 @@ LedgerDescriptor getLedgerForEntry(ByteBuf entry, final byte[] masterKey)
     }
 
     private Journal getJournal(long ledgerId) {
-        return journals.get(MathUtils.signSafeMod(ledgerId, journals.size()));
+        String dividendString = Double.toString(ledgerId);
+        byte[] secretBytes = null;
+        int index = 0;
+        try {
+            secretBytes = MessageDigest.getInstance("md5").digest(dividendString.getBytes());
+            BigInteger md5Code = new BigInteger(1, secretBytes);
```

Comment:
It looks to me that computing this value for every write to the journal is quite expensive.

Can we find something less resource consuming?

Reply:
+1 for the performance concern.

Would something like
```
final int PRIME = 31;
return journals.get(MathUtils.signSafeMod(ledgerId, PRIME * journals.size()));
```
work well enough?